src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -1175,6 +1150,149 @@ internal bool HasCollectionExpressionApplicableAddMethod(SyntaxNode syntax, Type
                 return result;
             }
 
+            static ImmutableArray<MethodSymbol> filterOutBadGenericMethods(
+                Binder addMethodBinder, SyntaxNode syntax, BoundMethodGroup methodGroup, MethodGroupResolution resolution,
+                ImmutableArray<MemberResolutionResult<MethodSymbol>> finalApplicableCandidates, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
+            {
+                Debug.Assert(methodGroup.ReceiverOpt is not null);
+
+                ImmutableArray<MethodSymbol> addMethods;
```

Comment:
>addMethods [](http://example.com/codeflow?start=45&length=10)

Consider moving `addMethods` declaration to `resultBuilder.ToImmutableAndFree()`, or remove the declaration if not needed.